conversions/c_atoi_str_to_integer.c

```diff
@@ -1,40 +1,87 @@
+/**
+ * \file
+ * \brief Recoding the original atoi function in stdlib.h 
+ * \author [Krishna Vedala](https://github.com/kvedala)
+ *
+ * The function convert a string passed to an integer 
+ */
 #include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
 
-// recoding the original atoi function in stdlib.h
-int	c_atoi(char *str)
+/**
+ * the function take a string and return an integer
+ * \param[out] str pointer to a char address
+ */
+int c_atoi(const char *str)
 {
-	int				i;
-	int				sign;
-	long			value;
-	long			prev;
-
-	i = 0;
-	sign = 1;
-	value = 0;
-	// skip wait spaces
-	while (((str[i] <= 13 && str[i] >= 9) || str[i] == 32) && str[i] != '\0')
-		i++;
-	// store the sign
-	if (str[i] == '-' || str[i] == '+')
-		(str[i++] == '-') ? sign = -1 : 1;
-	// convert char by char to an int value
-	while (str[i] >= 48 && str[i] <= 57 && str[i] != '\0')
-	{
-		prev = value;
-		value = value * 10 + sign * (str[i] - '0');
-		// manage the overflow
-		if (sign == 1 && prev > value)
-			return (-1);
-		else if (sign == -1 && prev < value)
-			return (0);
-		i++;
-	}
-	return (value);	
+    /**
```

Comment:
the comment style -
```c
/**
 * something something
 */
```
is reserved and should not be using within a function body. Inside a function body, you should use the normal comment blocks `/* --- */`

Reply:
alright i used to follow other rules :)